AndInTheClouds/chordreader2
# No Ticket: Resolve problems reported by Android Studio

# Introduction

This branch resolves a lot of "problems" that were reported by Android Studio, for example unused code and best practices.

In order to see these problems (e.g. on `master`), open the project in Android Studio. From the project explorer, right-click on the directory that you would like to scan (e.g. "app") and select "Analyze > Inspect code...". Under the "Problems" tab, you will see a report with all the issues that were found.

The branch does not fix _all_ of the issues, but a good bunch of them.

The main goal here is to have as little distraction as possible when reading/working with the code. A lot of the suggestions made by Android Studio are considered "Best Practices", so it makes sense to follow them.

I have tried to use small and atomic commits, which should make the review easier. (Rather than going through the _combined_ Git diff, you could go through the changes commit by commit.)

# How to test

Checkout this branch and deploy the app to your device (or run it in an emulator). The app should still be functioning as expected.
